File: VB/WindowsApplication3/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGauges.Core.Model
Imports DevExpress.XtraGauges.Win
Imports DevExpress.XtraGauges.Base
Imports DevExpress.XtraGauges.Win.Base
Imports DevExpress.XtraEditors


Namespace DxSample
	Partial Public Class MainForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub


		Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			timer1.Start()
			SetUpAnimation()
		End Sub

		Private Sub SetUpAnimation()
			arcScaleComponent1.EnableAnimation = True
			arcScaleComponent1.EasingFunction = New BackEase()
			arcScaleComponent1.EasingMode = DevExpress.XtraGauges.Core.Model.EasingMode.EaseInOut

			arcScaleComponent2.EnableAnimation = True
			arcScaleComponent2.EasingFunction = New CircleEase()
			arcScaleComponent2.EasingMode = EasingMode.EaseIn

			linearScaleComponent1.EnableAnimation = True
			linearScaleComponent1.EasingFunction = New QuadraticEase()
			linearScaleComponent1.EasingMode = EasingMode.EaseOut
		End Sub

		Private Sub OnTimerTick(ByVal sender As Object, ByVal e As EventArgs) Handles timer1.Tick
			DoAnimation(gaugeControl1)
		End Sub

		Private Sub DoAnimation(ByVal gauge As GaugeControl)
			For Each gb As IGauge In gauge.Gauges
				Dim cGauge As ICircularGauge = TryCast(gb, ICircularGauge)
				If cGauge IsNot Nothing Then
					For Each scale As IScale In cGauge.Scales
						scale.Value = AnimateScaleValue(scale)
					Next scale
				End If
				Dim lGauge As ILinearGauge = TryCast(gb, ILinearGauge)
				If lGauge IsNot Nothing Then
					For Each scale As IScale In lGauge.Scales
						scale.Value = AnimateScaleValue(scale)
					Next scale
				End If
			Next gb
		End Sub

		Private Function AnimateScaleValue(ByVal scale As IScale) As Single
			Dim r As New Random()
			Return r.Next(0, 100)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication3/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Drawing;
using DevExpress.XtraGauges.Core.Model;
using DevExpress.XtraGauges.Win;
using DevExpress.XtraGauges.Base;
using DevExpress.XtraGauges.Win.Base;
using DevExpress.XtraEditors;


namespace DxSample {
    public partial class MainForm : XtraForm
    {
        public MainForm() {
            InitializeComponent();
        }
      

        private void OnFormLoad(object sender, EventArgs e) {
            timer1.Start();
            SetUpAnimation();
        }

        private void SetUpAnimation()
        {
            arcScaleComponent1.EnableAnimation = true;
            arcScaleComponent1.EasingFunction = new BackEase();
            arcScaleComponent1.EasingMode = DevExpress.XtraGauges.Core.Model.EasingMode.EaseInOut;

            arcScaleComponent2.EnableAnimation = true;
            arcScaleComponent2.EasingFunction = new CircleEase();
            arcScaleComponent2.EasingMode = EasingMode.EaseIn;

            linearScaleComponent1.EnableAnimation = true;
            linearScaleComponent1.EasingFunction = new QuadraticEase();
            linearScaleComponent1.EasingMode = EasingMode.EaseOut;
        }

        void OnTimerTick(object sender, EventArgs e)
        {
            DoAnimation(gaugeControl1);
        }

        void DoAnimation(GaugeControl gauge)
        {
            foreach (IGauge gb in gauge.Gauges)
            {
                ICircularGauge cGauge = gb as ICircularGauge;
                if (cGauge != null)
                {
                    foreach (IScale scale in cGauge.Scales) scale.Value = AnimateScaleValue(scale);
                }
                ILinearGauge lGauge = gb as ILinearGauge;
                if (lGauge != null)
                {
                    foreach (IScale scale in lGauge.Scales) scale.Value = AnimateScaleValue(scale);
                }
            }
        }

        float AnimateScaleValue(IScale scale)
        {
            Random r = new Random();
            return r.Next(0, 100);
        }
    }
}
